include/eve/algo/concepts/iterator_cardinal.hpp

```diff
@@ -14,28 +14,42 @@
 namespace eve::algo
 {
   //================================================================================================
-  //! @addtogroup eve.algo.concepts
+  //! @addtogroup algo_concepts
   //! @{
   //!   @struct iterator_cardinal
   //!   @brief Returns a cardinal for an `eve::algo::iterator`
   //!
   //!   **Required header:** `#include <eve/algo/concepts.hpp>`
   //!
-  //!   For a given `eve::algo::iterator` returns a `cardinal`.
-  //!   In order to provide this, the `iterator` has to define a static method: `iterator_cardinal() -> fixed`
+  //!   For a given `eve::algo::iterator` type `I`, returns its associated `eve::cardinal` by
+  //!   computing the resturn type of `I::iterator_cardinal()`.
   //!
-  //!   This is often used as a helper to define iterators and because of how C++ works, it's useful to default
-  //!   to `expected_cardinal`
+  //!   This is often used as a helper to define iterators and because of how C++ works, it's
+  //!   recommended to default to `eve::expected_cardinal`.
```

Comment:
No, sorry.

The point here is that "if it's not  an `eve::iterator`, the function returns fixed<expected_cardinal_v<value_type_t<I>>>"

This makes meta-programming easier.

Reply:
This comment is still wrong